packages/@react-aria/gridlist/src/useGridListItem.ts

```diff
@@ -293,8 +323,25 @@ export function useGridListItem<T>(props: AriaGridListItemOptions, state: ListSt
   if (isVirtualized) {
     let {collection} = state;
     let nodes = [...collection];
```

Comment:
this means every grid item has to make a copy of the collection (O)n^2, can we do this at the useGrid level and send it (`hasSections`) via our hooks context or in the state?

Reply:
yeah we could add a new prop to `useGridListItem` called something `isInSection` and then determine that value inside of GridListItem by checking whether the parent node's type is a section. how does that sound?